variables.tf

```diff
@@ -904,3 +904,17 @@ variable "custom_container_apps" {
   }))
   default = {}
 }
+
+variable "container_app_identities" {
+  description = "Identities to assign to container app"
+  type = object({
+    type : string
+    identity_ids : list(string)
+  })
+  default = null
```

Comment:
Can you change the default data type to be the same as the expected data type? `default = {}` so it's consistent?

Reply:
Good shout, done!

Struggled with the validation a bit as a result, I would have expected 

`var.container_app_identities == [} ? true : contains(["SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned", ], container_app_identities.type)` to work, but it consistently complained about `.type` being null (even with the added `coalesce` statement I've added`. Similar/same problem with the other validation statement I've added.

I've tweaked them both, and they both work as expected for me where either:
- Both values can not be supplied
OR
- Both values must be supplied, but `type` must be one of three specific values, and `identity_ids` must contain at least one string

If you have a better suggestion/method of doing the validation for these I would be grateful to hear it! Not a Terraform expert by any means.

Reply:
Hm. I actually think moving back to `null` is the right move now I've thought about it some more..

```hcl
variable "container_app_identities" {
  description = "Identities to assign to the primary container app"
  type = object({
    type : string
    identity_ids : list(string)
  })
  default  = null

  validation {
    condition = var.container_app_identities != null ? alltrue([
      var.container_app_identities.type != "",
      contains(["SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned"], var.container_app_identities.type)
    ]) : true
    error_message = "Identity type must be one of the following: 'SystemAssigned', 'UserAssigned', or 'SystemAssigned, UserAssigned'!"
  }

  validation {
    condition     = var.container_app_identities != null ? length(var.container_app_identities.identity_ids) > 0 : true
    error_message = "At least one identity ID must be supplied"
  }
}
```

Reply:
Agreed, it makes things cleaner. Although I couldn't do it exactly this way as the null check doesn't work properly (I don't believe Terraform quits out of expressions early?), but I've implemented something similar.